commands/plugin/install.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+const spawn = require('child-process-ext/spawn');
+const fsp = require('fs').promises;
+const fse = require('fs-extra');
+const path = require('path');
+const _ = require('lodash');
+const isPlainObject = require('type/plain-object/is');
+const yaml = require('js-yaml');
+const cloudformationSchema = require('@serverless/utils/cloudformation-schema');
+const log = require('@serverless/utils/log');
+const ServerlessError = require('../../lib/serverless-error');
+const yamlAstParser = require('../../lib/utils/yamlAstParser');
+const fileExists = require('../../lib/utils/fs/fileExists');
+const npmCommandDeferred = require('../../lib/utils/npm-command-deferred');
+const CLI = require('../../lib/classes/CLI');
+const {
+  getPlugins,
+  getPluginInfo,
+  getServerlessFilePath,
+  validate,
+} = require('../../lib/commands/plugin-management');
+
+const cli = new CLI(undefined);
+
+module.exports = async ({ configuration, serviceDir, configurationFilename, options }) => {
+  const pluginInfo = getPluginInfo(options.name);
+  options.pluginName = pluginInfo[0];
+  options.pluginVersion = pluginInfo[1] || 'latest';
+
+  validate({ serviceDir });
+  const plugins = await getPlugins();
+  const plugin = plugins.find((item) => item.name === options.pluginName);
+  if (!plugin) {
+    cli.log('Plugin not found in serverless registry, continuing to install');
+  }
+
+  await pluginInstall({ configuration, serviceDir, options });
+  await addPluginToServerlessFile({ serviceDir, configurationFilename, options });
+  await installPeerDependencies({ serviceDir, options });
+
+  const message = [
+    'Successfully installed',
+    ` "${options.pluginName}@${options.pluginVersion}"`,
+  ].join('');
+  cli.log(message);
+};
+
+const pluginInstall = async ({ configuration, serviceDir, options }) => {
+  const packageJsonFilePath = path.join(serviceDir, 'package.json');
+
+  // check if package.json is already present. Otherwise create one
+  const exists = await fileExists(packageJsonFilePath);
+  if (!exists) {
+    cli.log('Creating an empty package.json file in your service directory');
+
+    const packageJsonFileContent = {
+      name: configuration.service,
+      description: '',
+      version: '0.1.0',
+      dependencies: {},
```

Comment:
AFAIK It's fine for `package.json` to not have all this fields, let's create only `devDependencies` field

Reply:
@medikoo It's even fine not to have `package.json` to run `npm install`. I think we created an empty `package.json` here to specify a service name. But I think that is not the responsibility of this command. Let me not create an empty `package.json`.

Reply:
Good call 👍